packages/std/src/addresses.rs

```diff
@@ -313,18 +313,50 @@ pub fn instantiate2_address(
     // Non-empty msg values are discouraged.
     // See https://medium.com/cosmwasm/dev-note-3-limitations-of-instantiate2-and-how-to-deal-with-them-a3f946874230.
     let msg = b"";
-    instantiate2_address_impl(checksum, creator, salt, msg)
+    let len = 32;
+    instantiate2_address_impl(checksum, creator, salt, msg, len)
 }
 
 /// The instantiate2 address derivation implementation. This API is used for
 /// testing purposes only. The `msg` field is discouraged and should not be used.
 /// Use [`instantiate2_address`].
+///
+/// `len` is the address length on bytes. The resulting address data will be truncated to
+/// that length. A value > 32 is invalid because [`hash`] returns only 32 bytes of data.
+/// A value of 0 is considered invalid because it indicates a bug.
```

Comment:
I was wondering if we should use a const generic for this because then we can validate this at compile time, but the drawback would be that you cannot specify the value at runtime. So I think it's fine like this with the debug asserts.

Reply:
Yes, but I could imagine people wanting to use the same wasm file for both cases and providing the address length in the instantiate message.

Reply:
Alright, so no strong benefits for either solution?

I just added a test case to mitigate the lack of Injective test data. Then we merge as is?

Reply:
Yes, let's merge as is, since both solutions have their benefits and drawbacks and it's not clear to me that the const version is necessarily better.